src/libmongoc/tests/test-mongoc-retryable-writes.c

```diff
@@ -568,6 +568,236 @@ test_unsupported_storage_engine_error (void)
+                 "'errorLabels': ['RetryableWriteError']}]}");
+
+   cleanup_original_error_test (client, server_id, &reply, coll, callbacks);
+}
+
+
+static void
+retryable_writes_original_error_find_modify (void *ctx)
+{
+   mongoc_client_t *client;
+   mongoc_collection_t *coll;
+   bson_t reply;
+   bson_error_t error = {0};
+   mongoc_apm_callbacks_t *callbacks = {0};
+   prose_test_3_apm_ctx_t apm_ctx = {0};
+   mongoc_find_and_modify_opts_t *opts;
+
+   BSON_UNUSED (ctx);
+
+   // setting up the client
+   client = test_framework_new_default_client ();
+   coll = get_test_collection (client, "retryable_writes");
+   callbacks = mongoc_apm_callbacks_new ();
+
+   // setup the test
+   const uint32_t server_id =
+      set_up_original_error_test (callbacks, &apm_ctx, "findAndModify", client);
+
+   // setup for findAndModify
+   bson_t query = BSON_INITIALIZER;
+   BSON_APPEND_UTF8 (&query, "x", "1");
+   bson_t *update = BCON_NEW ("$inc", "{", "x", BCON_INT32 (1), "}");
+   opts = mongoc_find_and_modify_opts_new ();
+   mongoc_find_and_modify_opts_set_update (opts, update);
+
+   // attempt a findAndModify operation
+   mongoc_collection_find_and_modify_with_opts (
+      coll, &query, opts, &reply, &error);
+
+   // assert error contains a writeConcernError with original error code
+   ASSERT_ERROR_CONTAINS (error, MONGOC_ERROR_WRITE_CONCERN, 91, "");
+
+   // the reply holds the original error information
+   ASSERT_MATCH (
+      &reply,
+      "{'lastErrorObject' : { 'n': 0, 'updatedExisting' : false }, 'value' : "
+      "null, 'writeConcernError' : { 'code': 91, 'errorLabels' : [ "
+      "'RetryableWriteError' ]}, 'ok' : 1.0}");
+
+   cleanup_original_error_test (client, server_id, &reply, coll, callbacks);
+   mongoc_find_and_modify_opts_destroy (opts);
+   bson_destroy (&query);
+   bson_destroy (update);
+}
+
+static void
+retryable_writes_original_error_general_command (void *ctx)
+{
+   mongoc_client_t *client;
+   mongoc_collection_t *coll;
+   bson_t reply;
+   bson_error_t error = {0};
+   mongoc_apm_callbacks_t *callbacks = {0};
+   prose_test_3_apm_ctx_t apm_ctx = {0};
+
+   BSON_UNUSED (ctx);
+
+   // setting up the client
+   client = test_framework_new_default_client ();
+   coll = get_test_collection (client, "retryable_writes");
+   callbacks = mongoc_apm_callbacks_new ();
+
+   // setup test
+   const uint32_t server_id =
+      set_up_original_error_test (callbacks, &apm_ctx, "insert", client);
+
+   bson_t *cmd = BCON_NEW ("insert",
+                           mongoc_collection_get_name (coll),
+                           "documents",
+                           "[",
+                           "{",
+                           "}",
+                           "]");
+
+   // attempt an insert operation
+   mongoc_client_command_simple (client, "admin", cmd, NULL, &reply, &error);
+
+   // writeConcernErrors are returned in the reply and not as an error
+   ASSERT_ERROR_CONTAINS (error, 0, 0, "");
```

Comment:
```suggestion
   // attempt an insert operation
   ASSERT (!mongoc_client_write_command_with_opts (
      client, "test", cmd, NULL, &reply, &error));

   // assert error contains a writeConcernError with original error code
   ASSERT_ERROR_CONTAINS (error, MONGOC_ERROR_WRITE_CONCERN, 91, "");
```
Unlike the other prose test functions, `mongoc_client_command_simple` does not fail as specified by the prose test description because it is not instructed that the command is a retryable write operation. The operation succeeds on first attempt (due to presence of `ok: 1`) and the `RetryableWriteError` label is not automatically actioned.

For consistency with the other prose test functions ("For the resulting error...") while still testing the `_mongoc_client_retryable_write_command_with_stream` code path, use `mongoc_client_write_command_with_opts` instead.